src/server/gc_worker/gc_manager.rs

```diff
@@ -514,14 +514,18 @@ impl<S: GcSafePointProvider, R: RegionInfoProvider + 'static, E: KvEngine> GcMan
             self.check_if_need_rewind(&progress, &mut need_rewind, &mut end);
 
             let controller: Arc<(Mutex<usize>, Condvar)> = Arc::clone(&task_controller);
-            let cb = Box::new(move |_res| {
-                let (lock, cvar) = &*controller;
-                let mut current_tasks = lock.lock().unwrap();
-                *current_tasks -= 1;
-                cvar.notify_one();
-                AUTO_GC_PROCESSED_REGIONS_GAUGE_VEC
-                    .with_label_values(&[PROCESS_TYPE_GC])
-                    .inc();
+            let cb = Box::new(move |res| {
+                if let Err(Error(box ErrorInner::GcWorkerTooBusy)) = res {
+                    // the task is actually not scheduled, so no action here.
+                } else {
+                    let (lock, cvar) = &*controller;
+                    let mut current_tasks = lock.lock().unwrap();
+                    *current_tasks -= 1;
```

Comment:
In line 608 
https://github.com/zyguan/tikv/blob/9e0c123dd7f303c61c2b519d1169127e2069ed03/src/server/gc_worker/gc_manager.rs#L607-L611

It seems that, the counter incr by 1 no matter whether there is error or not. If we do not to acquire the lock and deal with counter here, then we probably also need to check the result there, and incr the counter conditionally.